autometrics/src/labels.rs

```diff
@@ -246,3 +228,75 @@ pub trait GetStaticStr {
     }
 }
 impl_trait_for_types!(GetStaticStr);
+
+#[macro_export]
+macro_rules! result_labels {
+    ($e:expr) => {{
+        use $crate::__private::{
+            GetLabels, GetStaticStr, ResultAndReturnTypeLabels, ERROR_KEY, OK_KEY,
+        };
+        $crate::__private::spez! {
```

Comment:
Wherever this code lives, it would be worth putting a comment along the lines of:
> We are using the [autoref specialization](https://github.com/dtolnay/case-studies/blob/master/autoref-specialization/README.md) trick to get the labels for the result type in a variety of circumstances. Specifically, if the function returns a `Result`, we will add the `ok` or `error` label accordingly _unless_ one or both of the types that the `Result<T, E>` is generic over implements the `GetLabels` trait.